internal/pkg/instrumentation/manager.go

```diff
@@ -306,6 +308,26 @@ func (m *Manager) Run(ctx context.Context, target *process.TargetDetails) error
 	return <-done
 }
 
+// Stop stops all probes and cleans up all the resources associated with them.
+func (m *Manager) Stop() error {
+	if m.state == managerStateUninitialized || m.state == managerStateStopped {
+		return nil
+	}
+
+	m.probeMu.Lock()
+	defer m.probeMu.Unlock()
+
+	m.logger.Debug("Shutting down all probes")
+	err := m.cleanup(m.td)
+
+	// Wait for all probes to stop before closing the chan they send on.
+	m.runningProbesWG.Wait()
+	close(m.telemetryCh)
+
+	m.state = managerStateStopped
+	return err
+}
+
 func (m *Manager) load(target *process.TargetDetails) error {
```

Comment:
It seems a bit odd to have both `Load` and `load`. Should these methods be unified?

Reply:
renamed the internal function to `loadProbes`